piqe_ocp_lib/api/resources/ocp_resource_quotas.py

```diff
@@ -0,0 +1,926 @@
+        except ApiException as e:
+            logger.exception(f"Exception while replacing ResourceQuota definitions : {e}\n")
+
+        return replace_rq_response
+
+    def get_a_resource_quota_status(self, name: str, namespace: str) -> V1ResourceQuotaStatus:
+        """
+        Query a ResourceQuota in specific namespace for status usage
+        :param name: (str) name of ResourceQuota
+        :param namespace: (str) namespace where ResourceQuota is created
+        :return: V1ResourceQuotaStatus object
+        """
+
+        """
+        The dynamic client builds subresource APIs for resources as well.
+        In this case ResourceQuota has a Subresource called Status that represents
+        the ResourceQuotaStatus API to see usage statistics.
+        
+        "kind": "ResourceQuota",
+        "name": "resourcequotas/status"
+        """
+        resp = self.ocp_resource_quota.status.get(name=name, namespace=namespace)
+        status = resp.status
+        return V1ResourceQuotaStatus(**status)
+
+
+
+    def resource_scope_selector_expression(self) -> Union['OcpScopeSelectorExpression',
+                                                          V1ScopedResourceSelectorRequirement]:
+        """
+        Create a OcpScopeSelectorExpression object that is used to build a V1ScopedResourceSelectorRequirement.
+        For more info about scopeSelectors refer to
+        https://kubernetes.io/docs/concepts/policy/resource-quotas/#resource-quota-per-priorityclass
+
+        #Example:
+        item = ocp_rq.resource_scope_selector_expression().priority_class_scope()._in().values(vals=['high']).build()
+
+        :return: OcpScopeSelectorExpression object
+        """
+
+        return OcpScopeSelectorExpression()
+
+    def resource_quota_spec(self) -> Union['OcpStorageQuotaSpec', V1ResourceQuotaSpec]:
+        """
+        Create an OcpResourceQuotaSpec object that is used to build a V1ResourceQuotaSpec
+
+        #Example 1:
+        item = ocp_rq.resource_quota_spec().memory(requests='512Mi').cpu(requests='1200M').terminating_scope().build()
+
+        #Example 2:
+        item = ocp_rq.resource_quota_spec().ephemeral_storage(requests="1G").storage_class(class_name='gold',
+                                                                                           requests='8G').build()
+
+        :return: OcpPodLimit object to build resource quotas
+        """
+
+        return OcpResourceQuotaSpec()
+
+    def resource_quota(self, name: str, namespace: str, spec: V1ResourceQuotaSpec) -> V1ResourceQuota:
+        """
+        Build an V1ResourceQuota Object using the V1ResourceQuotaSpec
+        :return:  V1ResourceQuota Object
+        """
+
+        return V1ResourceQuota(api_version='v1',
+                            kind='ResourceQuota',
+                            metadata=V1ObjectMeta(name=name, namespace=namespace),
+                            spec=spec)
+
+
+class AutoEnum(Enum):
```

Comment:
I'd like to share a dirty trick over here cause it might be helpful:
```python
class StrEnum(str, Enum):
    foo = 'bar'
    __str__ = str.__str__ 
    # or approach below if you want it explicit
    def __str__(self):
        return self.value  
```

Now the comparison StrEnum.foo == 'bar' matches. No need to access `.value`